docs/Brew-Test-Bot-For-Core-Contributors.md

```diff
@@ -1,23 +1,26 @@
-# Brew Test Bot for Maintainers
+# BrewTestBot for Maintainers
 
-If a build has run and passed on `brew test-bot` then it can be used to quickly bottle formulae.
+`brew test-bot` is the command our CI runs to test and build bottles for formulae.
 
-## Bottling
+## Publishing Bottles
 
-If a pull request is correct and doesn't need any modifications to commit messages or otherwise:
+If CI is passing on a pull request and it doesn't need any modifications (i.e. commit message, revision bump, etc.):
 
 1. Review and approve the pull request. Be sure to thank the contributor!
-2. Wait for BrewTestBot to automatically merge the pull request. This happens about once an hour. BrewTestBot will comment if there is a failure.
+2. Wait for BrewTestBot to automatically merge the pull request. This job usually starts within a minute if both of the following are true:
+    - The pull request is approved by a maintainer who has write access to homebrew-core.
+    - CI is passing, and all jobs are green.
+    
+If any jobs did not complete successfully, the pull request will not automatically merge. Additionally, BrewTestBot will comment on the pull request if there is a publishing failure.
 
-If a pull request won't be automatically merged by Brew Test Bot (has the labels `do not merge`, `new formula`, or `automerge-skip`), but the [commit messages and commit style](Formula-Cookbook.md#commit) are correct:
+If a pull request won't be automatically merged by BrewTestBot (has the labels `autosquash`, `automerge-skip`, or`new formula`, or has some kind of acceptable CI failure):
```

Comment:
The goal is to never have failing CI, and I assume we don't have documented what's an „acceptable CI failure”, so I don't think it is useful to mention here.

Reply:
I believe new formula require manual intervention at the moment

Reply:
I agree that having failing CI is never ideal, but in this case I'm thinking about things like: checksum mismatches that have been verified, new formula that we are waiving the notability requirement for, etc. The alternative would be setting up flags for BrewTestBot to ignore all the different types of "acceptable" failures we currently see.

Reply:
> The alternative would be setting up flags for BrewTestBot to ignore all the different types of "acceptable" failures we currently see.

I mean that's how it already works for casks using `ci-*` labels.

Reply:
Right, and we have some of that in core. I just don't think it's currently as comprehensive. @MikeMcQuaid has been prodding us to move toward a "no red CI" approach, so this may be the time to figure out any remaining instances of this.

Reply:
There's still quite a bit more to do, like how to handle a flaky audit in a long multi-day build (running the whole thing again is a waste of time) as well as dependent testing which we've not touched yet at all.

Reply:
> The alternative would be setting up flags for BrewTestBot to ignore all the different types of "acceptable" failures we currently see.

I think we should do this. It's pretty straightforward now and I can help get documentation updated for this.

> There's still quite a lot more to do, like how to handle a flaky audit in a long multi-day build (running the whole thing again is a waste of time)

This may be a good candidate for a "repo admins can merge" branch protection and we limit this to TSC.

> as well as dependent testing which we've not touched yet at all.

iirc (but @carlocab can confirm) I don't think `conclusion` fails if dependent jobs do?

Reply:
This all sounds good to me. Can also confirm that dep testing failures do not cause `conclusion` to fail.

Reply:
> This may be a good candidate for a "repo admins can merge" branch protection and we limit this to TSC.

Also: we should consider pulling these "flaky audits" into a separate job that can be non-required.

Reply:
Also also: if we have known flaky audits that are unfixed: this should be tracked in an issue.

Reply:
There's not really known flaky per se but more along the lines of say the formula homepage (but not source) temporarily goes down during a long build, which is something that's silly to perform a rebuild over. The nature of the internet means these things happen.

Reply:
Yeh, I think we should consider pulling all these sorts of audits into another job (also because they are not OS dependent).